Upgraded Storybook to v9 in Shade

peterzimon opened the pull request:
ref https://linear.app/ghost/issue/PROD-2644/upgrade-storybook-to-v9

A Storybook addon were automatically [updated](https://github.com/TryGhost/Ghost/commit/67298abcbca68b13782963037106e591c4eda5e4) by Renovate, which made Storybook break when ran locally. To overcome this and future compatible issues in Shade we upgraded Storybook to its latest version.

- The upgrade was executed by running the upgrade script `npx storybook@latest upgrade` ([docs](https://storybook.js.org/docs/releases/upgrading)) which updated all storybook components and stories to use the Storybook `react-vite` module instead of `react`.
- Some Storybook addons (essentials, interactions, links, styling, blocks) were incompatible with v9 so they were removed from `package.json`. We'll add new addons in separate PRs if needed.
- `glob` v10+ is ESM-only and doesn't provide a default export so the import in Vite config was updated accordingly.

peterzimon committed:
Upgraded Storybook to v9

ref https://linear.app/ghost/issue/PROD-2644/upgrade-storybook-to-v9

- Some Storybook addons were automatically updated by Renovate. These are not compatible anymore with the version we're using in Shade so we had to upgrade to the latest version of Storybook.

peterzimon committed:
Updated formatting

peterzimon committed:
Cleanup

peterzimon committed:
Cleanup

peterzimon committed:
Reverted config

peterzimon committed:
Fixed unit test

ErisDS commented on apps/shade/.storybook/main.tsx:
```diff
@@ -1,30 +1,33 @@
-import path from 'path';
+import path, { dirname, join } from 'path';
+
+const require = createRequire(import.meta.url);
```
I understand that these changes - the createRequire and dirname stuff are all to do with ESM vs commonJS modules, but it's weird to me this worked before and doesn't now. Not entirely sure why this belongs with this commit or if there's an easier way to resolve these things?

peterzimon replied:
I assume these appear now because the Storybook dependencies are upgraded (e.g. `glob` is now on v10 in Shade).

ErisDS reviewed (commented):
I've left a couple of questions about what's in the commit, but I don't see anything majorly wrong to block merge

peterzimon commented on apps/shade/.storybook/main.tsx:
```diff
@@ -1,30 +1,33 @@
 };
 export default config;
+
+function getAbsolutePath(value: string): any {
```
Nope, it was added by the upgrade script. I'll take a look if this can be eliminated

peterzimon replied:
I removed `getAbsolutePath`. It was added by the script but was unnecessary in our setup.

peterzimon committed:
Removed getAbsolutePath function

peterzimon committed:
Revert using getAbsolutePath

peterzimon committed:
Removed unnecessary path function

peterzimon committed:
Removed double dependencies

rollup-plugin-node-builtins module was added to both dev and prod dependencies.

peterzimon committed:
Updated yarn.lock

peterzimon committed:
Removed unused require from main storybook component

peterzimon committed:
Removed unused import

peterzimon committed:
Import React namespace in color picker story

peterzimon committed:
Fixed color picker Storybook interactions panel bug

peterzimon committed:
Merge branch 'main' into storybook-v9-upgrade

peterzimon committed:
Updated color picker stories and component
